cmd/storagenode/deprecated.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// Deprecated contains deprecated config structs
+type Deprecated struct {
+	Kademlia struct {
+		ExternalAddress string `default:"" hidden:"true"`
+		Operator        struct {
+			Email  string `default:"" hidden:"true"`
+			Wallet string `default:"" hidden:"true"`
+		}
+	}
+}
+
+// maps deprecated config values to new values if applicable
+func mapDeprecatedConfigs(log *zap.Logger) {
+	type migration struct {
+		newValue        interface{}
```

Comment:
Tips: Since it's only using strings you wouldn't need to use interfaces.

Also you can avoid the reflection by using interface for the mapping types instead. See https://egonelbre.com/manipulating-json-fields/ for something similar.